src/vscode-bicep/src/commands/generateParams.ts

```diff
@@ -27,19 +27,31 @@ export class GenerateParamsCommand implements Command {
 
     try {
       console.log(`Generating parameters file for ${documentUri.fsPath}...`);
+
+      const outputFormat = await vscode.window.showQuickPick(['json', 'bicepparam'], { title: "Please select the output format" });
+      const includeParams = await vscode.window.showQuickPick(['requiredonly', 'all'], { title: "Please select which parameters to include" });
+
+      if (outputFormat === undefined || includeParams === undefined) {
+        throw new Error("Please select the output format and which parameters to include");
```

Comment:
[nit] I am thinking user experience wise, this condition won't be true (since user must pick valid option from the list) but if it does happen, should we produce error message like "Incorrect output format or Include params value" instead of prompting the user again?

Reply:
actually, if the user presses `Esc` , when they prompted to pick the output format and which parameters to include, `outputFormat` and `includeParams` variables have `undefined` value.

![image](https://github.com/Azure/bicep/assets/118744/1e661893-6ca2-4954-9494-fee61f0bc8b8)

I also think there is an improvement opportunity with the message, but it'd be better if we can inform the user about the issue, and don't smell so technical.

![image](https://github.com/Azure/bicep/assets/118744/0680bb04-66a5-4fcc-994a-fe686bb3831e)

what you think?

Reply:
I see so users can cause both values to be `undefined` 

If we look at the wording of the message as `<Error>: <Suggestion>` rather `<Error>: <Prompt>` then it makes sense so I think it fine to keep the same message